src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -227,9 +235,57 @@ private SwiftCompileStep makeCompileStep(SourcePathResolver resolver) {
         "-emit-module-path",
         modulePath.toString(),
         "-emit-objc-header-path",
-        headerPath.toString(),
+        headerPath.toString()
+      );
+
+    if (shouldGenerateIndex) {
+      compilerCommand.add(
+        "-num-threads",
+        "12",
+        "-index-store-path",
+        "buck-out"
+      );
+
+      // If there are too many output files, we need to use a output file list instead.
+      // Otherwise, we would get an error about command line too long.
+      boolean use_output_filelist = false;
+      PrintWriter output_filelist = null;
+      String output_filelist_path = "/tmp/" + moduleName + "-output-filelist";
+      if (srcs.size() > MAX_OUTPUT_ARG_COUNT) {
```

Comment:
Why don't we always use output_filelist?

Reply:
Because that would require additional IO for writing and reading the file and affects the performance.